web-app/src/components/BookingModal/BookingModalForm/container.js

```diff
@@ -1,35 +1,58 @@
 import React from 'react';
 import { PropTypes as PT } from 'prop-types';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import {
+  toggleBookingModal,
+  selectBooking,
+  updateEventDuration,
+} from '../../../actions/dashboard';
 import Swal from 'sweetalert2';
+import moment from 'moment';
 import {
   updateHoliday,
   requestHoliday,
   rejectHoliday,
 } from '../../../services/holidayService';
+import { eventBeingUpdated } from '../../../reducers';
 
-export default Wrapped =>
+const Container = Wrapped =>
   class extends React.Component {
     static propTypes = {
       employeeId: PT.number.isRequired,
       updateTakenHolidays: PT.func.isRequired,
-      updateBookingAndDuration: PT.func.isRequired,
-      closeModal: PT.func.isRequired,
+      selectBooking: PT.func.isRequired,
+      toggleModal: PT.func.isRequired,
       booking: PT.object.isRequired,
+      isEventBeingUpdated: PT.bool,
+      updateEventDuration: PT.func,
+      bookingDuration: PT.number,
     };
 
     constructor(props) {
       super(props);
       this.state = {
         formData: {
-          end: props.booking.end,
-          eventTypeId: props.booking.eventType.eventTypeId,
-          isHalfday: props.booking.halfDay || false,
-          start: props.booking.start,
+          end: moment(),
+          eventTypeId: 1,
+          isHalfday: false,
+          start: moment(),
         },
         formIsValid: true,
       };
     }
 
+    componentDidMount = () => {
+      this.setState({
```

Comment:
Just my own thought. But seeing a lot of `this.props...`. I think it would be neater to do ` const { booking: { halfDay, start, eventType }} = this.props;`